crates/sui-core/src/authority/authority_store.rs

```diff
@@ -690,14 +721,15 @@ impl AuthorityStore {
     pub fn check_sequenced_input_objects(
         &self,
         digest: &TransactionDigest,
-        objects: &[InputObjectKind],
+        objects: &mut Vec<InputObjectKind>,
```

Comment:
Please avoid passing mut vectors as much as possible. You can return new vector if necessary

Reply:
happy to update, curious about the reason?

Reply:
It makes the code easier to reason about